services/app-api/src/postgres/contractAndRates/submitContract.ts

```diff
@@ -77,8 +69,28 @@ async function submitContractInsideTransaction(
         }
     }
 
-    // this is all the revs that the newly submitted contract will be connected to. Those to be submitted and those already submitted.
-    const draftRateRevs = unsubmittedChildRevs.concat(linkedRateRevs)
+    const submissionResult = await submitContractAndRates(
+        tx,
+        contractID,
+        unsubmittedChildRevs.map((r) => r.rateID),
+        submittedByUserID,
+        submittedReason
+    )
+    if (submissionResult instanceof Error) {
+        return submissionResult
+    }
+
+    return await findContractWithHistory(tx, contractID)
+}
+
+async function submitContractAndRates(
+    tx: PrismaTransactionType,
+    contractID: string | undefined,
```

Comment:
So before we had an api almost working with editing and submitting a rate independently. Back when we did rate unlock and edit.


Kind of feels like we walked back from that and doubled down on the package idea. Are you thinking we just reimplemented a new function for independent rate editing?

Reply:
No, this supports that. The submitRate function calls this passing an undefined contractID, and it sets all the package submission stuff correctly. My goal was for all that bookkeeping to happen in one place, but submitRate works great with it on its own. And yes, I can move this to its own file to make that more clear.